dart/dynamics/SmartPointer.h

```diff
@@ -59,7 +61,12 @@ namespace dynamics {
   typedef std::weak_ptr< X >        Weak ## X ## Ptr;       \
   typedef std::weak_ptr< const X >   WeakConst ## X ## Ptr;
 
-// Skeleton smart pointers
+class SimpleFrame;
+DART_DYNAMICS_MAKE_SHARED_WEAK(SimpleFrame)
```

Comment:
It's not directly related to this PR but we could save some lines by putting the class declaration into the macros like:
```cpp
#define DART_DYNAMICS_MAKE_SHARED_WEAK( X )                 \
  class X ;
  typedef std::shared_ptr< X >      X ## Ptr;               \
  typedef std::shared_ptr< const X > Const ## X ## Ptr;     \
  typedef std::weak_ptr< X >        Weak ## X ## Ptr;       \
  typedef std::weak_ptr< const X >   WeakConst ## X ## Ptr;

// class SimpleFrame;  // Don't need anymore
DART_DYNAMICS_MAKE_SHARED_WEAK(SimpleFrame)
```